addons/mail/static/src/models/message/message.js

```diff
@@ -420,11 +420,33 @@ function factory(dependencies) {
         }
 
         /**
+         * The method does not attempt to cover all possible cases of empty
+         * messages, but mostly those that happen with a standard flow. Indeed
+         * it is preferable to be defensive and show an empty message sometimes
+         * instead of hiding a non-empty message.
+         *
+         * The main use case for when a message should become empty is for a
+         * message posted with only an attachment (no body) and then the
+         * attachment is deleted.
+         *
+         * The main use case for being defensive with the check is when
+         * receiving a message that has no textual content but has other
+         * meaningful HTML tags (eg. just an <img/>).
+         *
          * @private
+         * @returns {boolean}
          */
         _computeIsEmpty() {
             return (
-                (!this.body || htmlToTextContentInline(this.body) === '') &&
+                (
+                    !this.body ||
+                    [
+                        '',
+                        '<p></p>',
+                        '<p><br></p>',
+                        '<p><br/></p>',
+                    ].includes(this.body.replace(/\s/g, ''))
+                ) &&
                 this.attachments.length === 0 &&
                 this.tracking_value_ids.length === 0 &&
                 !this.subtype_description
```

Comment:
I don't like long conditional statements with `||`.
Feels way too academic and hard to read, especially when it's incorporated in a longer conditional statement containing `&&`.

I prefer you define a boolean and local variable for what is stated as empty body.
JS sucks a bit when you want to scope computation of a local variable, and also when pattern matching features is only old `switch/case`... But the idea is so:
```js
const isBodyEmpty = (() => {
    if (!this.body) {
        return true;
    }
    const bodyWithoutSpaces = this.body.replace(/\s/g, '');
    switch (bodyWithoutSpaces) {
        case '':
        case '<p></p>':
        case '<p><br></p>':
        case '<p><br/></p>':
            return true;
        default:
            return false;
    }
})();

return (
    isBodyEmpty &&
    [...]
);
```

It's longer, but I feel it's less cryptic overall and easier to understand at a glance, while being also easier to maintain. What do you think?

_(just to be clear, this comment is really a very small deal and more random ideas)_

Reply:
I understand what you mean and I agree (I don't like conditions with `or` in general and especially mixed with `and`) but I feel like introducing a function for that doesn't make it more readable.

I would prefer in that case just make 1 variable for the `or` part (and depending on what we need, could even be its own field) and then use that inside the `and` part.

Reply:
Done the split with just a variable. It's a bit better. If we really want a separate function, I would make a new field instead.